expression/builtin.go

```diff
@@ -87,10 +88,32 @@ func (b *baseBuiltinFunc) collator() collate.Collator {
 	return b.ctor
 }
 
-func newBaseBuiltinFunc(ctx sessionctx.Context, funcName string, args []Expression, retType types.EvalType) (baseBuiltinFunc, error) {
+func adjustNullFlagForReturnType(funcName string, args []Expression, bf baseBuiltinFunc) {
```

Comment:
adjustNullFlagForReturnTypeAfterWrapCast ? ...

Reply:
But build `cast` function will call this function too.